### PR TITLE
DON-1088 – safer input selector checks

### PR DESCRIPTION
**Pull request by NoelLH**
And a few other small improvements

### CODE REVIEW

File: support/check.js [hunk 1]
```diff
@@ -80,6 +80,31 @@ export async function checkTitle(title, seconds = WAIT_SECONDS) {
...
+ * @param {WebdriverIO.Element} element Already-located element.
+ * @param {string} value
+ * @param {number} seconds Number of seconds to wait.
+ * @returns {Promise<any>} `waitUntil()` result.
+ */
+async function checkValue(element, value, seconds = WAIT_SECONDS) {
+    return browser.waitUntil(
+        async () => {
+            const text = element.getValue();
+            if ((await text) !== value) {
```
Comment (bdsl) [inline support/check.js:95]:
```suggestion
            const text = await element.getValue();
            if (text !== value) {
```

File: pages/DonateStartPage.js [hunk 2]
```diff
@@ -190,12 +191,7 @@ export default class DonateStartPage {
      *
      */
     async checkDonorFirstName(expectedFirstName) {
-        const firstName = await $(firstNameSelector).getValue();
-        if (firstName !== expectedFirstName) {
-            throw new Error(
-                `First name value not as expected. Expected "${expectedFirstName}", got "${JSON.stringify(firstName)}"`
-            );
-        }
+        await checkSelectorValue(firstNameSelector, expectedFirstName);
```
Comment (bdsl) [inline pages/DonateStartPage.js:196]: nice reduction in code duplication. Might be worth going further and inlining this and similar methods now that the implementation isn't much more complicated than the name.

image
image

**Review by bdsl — APPROVED** [review]
LGTM, added one suggestion for a further improvement.

File: pages/DonateStartPage.js [hunk 3]
```diff
@@ -113,6 +114,8 @@ export default class DonateStartPage {
      * @param {string} expectedText  Text anticipated somewhere in the box.
      */
     async checkIdInfo(expectedText) {
+        // eslint-disable-next-line wdio/no-pause
+        await browser.pause(1500); // Give the login call a short processing time.
```
Comment (bdsl) [inline pages/DonateStartPage.js:118]: not a blocker just curious if you know if there's any difference in effect between doing this or doing `import { setTimeout } from 'node:timers/promises'; await setTimeout(1500);`.

File: pages/DonateStartPage.js (hunk 2 above)
Comment (NoelLH) [reply pages/DonateStartPage.js:196]: @bdsl possibly, we do use the same selector constant in `inputSelectorValue()` though, so I'm not sure we save a lot from inlining overall given we'd then need that referenced across 2 files instead

**Pull request merged**